firmware/src/services/iot/cloud/cloud_service.c

```diff
@@ -0,0 +1,628 @@
+            }
+            else if (shared_networking_params.haveAPConnection == 0)
+            {
+                // No network yet.
+                debug_printTrace("CLOUD: Waiting for WiFi AP Connection");
+            }
+            else if (shared_networking_params.haveIpAddress == 0)
+            {
+                // No IP yet.
+                debug_printInfo("CLOUD: Waiting for DHCP IP Address");
+            }
+            //else if (shared_networking_params.haveHostIp == 0)
+            else if (shared_networking_params.haveHostIp == 0)
+            {
+                // Need IP Address of MQTT Host to connect socket.
+                if (dnsRetryCount > 0)
+                {
+                    // still waiting for DNS look up
+                    dnsRetryCount--;
+                    break;
+                }
+                else
+                {
+                    // send request to get Host IP
+                    debug_printInfo("CLOUD: Getting IP for %s", mqtt_host);
+                    if (gethostbyname((char*)mqtt_host) != M2M_SUCCESS)
+                    {
+                        debug_printError("CLOUD: gethostbyname failed");
+                    }
+                    else
+                    {
+                        dnsRetryCount = DNS_RETRY_COUNT;
+                    }
+                }
+            }
+            else if (MQTT_GetConnectionState() == CONNECTED)
+            {
+                CLOUD_reset();
+            }
+            else
+            {
+                // Ready to connect socket
+                assert(shared_networking_params.haveHostIp == 1);
+
+                *mqttConnnectionInfo->tcpClientSocket = BSD_socket(PF_INET, BSD_SOCK_STREAM, 1);   // WINC_TLS
+
+                if (*mqttConnnectionInfo->tcpClientSocket >= 0)
+                {
+                    packetReceptionHandler_t* sockInfo = getSocketInfo(*mqttConnnectionInfo->tcpClientSocket);
+                    if (sockInfo != NULL)
+                    {
+                        sockInfo->socketState = SOCKET_CLOSED;
+                    }
+                    else
+                    {
+                        debug_printWarn("CLOUD: Socket Info Null");
+                    }
+                }
+            }
+            break;
+        }
+        case SOCKET_CLOSED:   // 1
+        {
+            // Connect to socket
+            if (connectMQTTSocket() != BSD_SUCCESS)
+            {
+                debug_printError(" CLOUD: Failed to connect socket");
+            }
+            break;
+        }
+        case SOCKET_IN_PROGRESS:   //2
+            break;
+
+        case SOCKET_CONNECTED:   // 3
+        {
+
+            mqttCurrentState mqttState = MQTT_GetConnectionState();
+
+            // typedef enum
+            // {
+            //     DISCONNECTED   = 0,
+            //     CONNECTING     = 1,
+            //     WAITFORCONNACK = 2,
+            //     CONNECTED      = 3
+            // } mqttCurrentState;

```

Comment:
needed?

Reply:
Again... a note/comment for debugging purpose.  Removed.